src/lib.rs

```diff
@@ -458,18 +458,37 @@ impl FastText {
         Ok(())
     }
 
-    pub fn predict(&self, text: &str, k: i32, threshold: f32) -> Vec<Prediction> {
-        let c_text = CString::new(text).unwrap();
+    fn convert_predictions(c_preds: &[fasttext_prediction_t]) -> Vec<Prediction> {
         unsafe {
-            let ret = cft_fasttext_predict(self.inner, c_text.as_ptr(), k, threshold);
-            let c_preds = slice::from_raw_parts((*ret).predictions, (*ret).length);
             let preds: Vec<Prediction> = c_preds.iter().map(|p| {
                 let label = CStr::from_ptr((*p).label).to_string_lossy().to_string();
                 Prediction {
                     prob: (*p).prob,
                     label: label
                 }
             }).collect();
+            preds
+        }
+    }
+
+    pub fn predict(&self, text: &str, k: i32, threshold: f32) -> Vec<Prediction> {
+        let c_text = CString::new(text).unwrap();
+        unsafe {
+            let ret = cft_fasttext_predict(self.inner, c_text.as_ptr(), k, threshold);
+            let c_preds = slice::from_raw_parts((*ret).predictions, (*ret).length);
+            let preds = FastText::convert_predictions(c_preds);
+            cft_fasttext_predictions_free(ret);
+            preds
+        }
+    }
+
+    pub fn predict_on_words(&self, words: &Vec<i32>, k: i32, threshold: f32) -> Vec<Prediction> {
```

Comment:
Prefer slice over reference to a `Vec`: `words: &[i32]`

Reply:
I don't quite understand how to get a `words: &Vec<i32>` to pass to this method. Can you add a test case?

Reply:
done. Worth noting, that such method only works with known tokens. Otherwise fasttext return `-1` as word id and predict method would fail then.

It looks like there is API to get vectors for unknown words, but no way to pass vectors into predict methid. Fasttext C++ API exposes only two:

1.  predict on sentence
2. predict on word ids